Functions/FirebaseFunctions/FIRFunctions.m

```diff
@@ -97,33 +99,47 @@ + (instancetype)functionsForApp:(FIRApp *)app {
 }
 
 + (instancetype)functionsForRegion:(NSString *)region {
-  return [[self alloc] initWithApp:[FIRApp defaultApp] region:region];
+  return [[self alloc] initWithApp:[FIRApp defaultApp] region:region customDomain:nil];
+}
+
++ (instancetype)functionsForCustomDomain:(NSString *)customDomain {
+  return [[self alloc] initWithApp:[FIRApp defaultApp]
+                            region:kFUNDefaultRegion
+                      customDomain:customDomain];
 }
 
 + (instancetype)functionsForApp:(FIRApp *)app region:(NSString *)region {
-  return [[self alloc] initWithApp:app region:region];
+  return [[self alloc] initWithApp:app region:region customDomain:nil];
 }
 
-- (instancetype)initWithApp:(FIRApp *)app region:(NSString *)region {
++ (instancetype)functionsForApp:(FIRApp *)app customDomain:(NSString *)customDomain {
+  return [[self alloc] initWithApp:app region:kFUNDefaultRegion customDomain:customDomain];
+}
+
+- (instancetype)initWithApp:(FIRApp *)app
+                     region:(NSString *)region
+               customDomain:(NSString *)customDomain {
   return [self initWithProjectID:app.options.projectID
                           region:region
+                    customDomain:customDomain
                             auth:FIR_COMPONENT(FIRAuthInterop, app.container)
                        messaging:FIR_COMPONENT(FIRMessagingInterop, app.container)];
 }
 
 - (instancetype)initWithProjectID:(NSString *)projectID
                            region:(NSString *)region
+                     customDomain:(NSString *)customDomain
                              auth:(nullable id<FIRAuthInterop>)auth
                         messaging:(nullable id<FIRMessagingInterop>)messaging {
   self = [super init];
   if (self) {
     if (!region) {
-      FUNThrowInvalidArgument(@"FIRFunctions region cannot be nil.");
+      FUNThrowInvalidArgument(@"FIRFunctions region cannot nil.");
     }
     _fetcherService = [[GTMSessionFetcherService alloc] init];
     _projectID = [projectID copy];
-    _region = [region copy];
-    _serializer = [[FUNSerializer alloc] init];
+    _region = [region copy] _customDomain = [customDomain copy] _serializer =
+        [[FUNSerializer alloc] init];
```

Comment:
This is either a typo or needs to be made clearer.

Reply:
Typo!